File: services/user-service/src/services/email_service.py
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart
import os
from utils.logger import logger

class EmailService:
    def __init__(self):
        self.smtp_server = os.getenv("SMTP_SERVER", "localhost")
        self.smtp_port = int(os.getenv("SMTP_PORT", "587"))
        self.smtp_username = os.getenv("SMTP_USERNAME", "")
        self.smtp_password = os.getenv("SMTP_PASSWORD", "")
        self.from_email = os.getenv("FROM_EMAIL", "noreply@ecommerce.com")

    async def send_verification_email(self, to_email: str, first_name: str, verification_token: str):
        """Send email verification email"""
        try:
            subject = "Verify Your Email Address"
            
            # Create verification URL
            verification_url = f"http://localhost:3000/verify-email?token={verification_token}"
            
            # HTML email content
            html_content = f"""
            <html>
                <body>
                    <h2>Welcome to E-commerce Platform!</h2>
                    <p>Hi {first_name},</p>
                    <p>Thank you for registering with us. Please click the link below to verify your email address:</p>
                    <p><a href="{verification_url}" style="background-color: #4CAF50; color: white; padding: 10px 20px; text-decoration: none; border-radius: 5px;">Verify Email</a></p>
                    <p>If the button doesn't work, copy and paste this link into your browser:</p>
                    <p>{verification_url}</p>
                    <p>This link will expire in 24 hours.</p>
                    <p>Best regards,<br>E-commerce Platform Team</p>
                </body>
            </html>
            """
            
            # For development, just log the email instead of sending
            logger.info(f"Email verification sent to {to_email}")
            logger.info(f"Verification URL: {verification_url}")
            
            return True
            
        except Exception as e:
            logger.error(f"Failed to send verification email: {e}")
            return False

    async def send_password_reset_email(self, to_email: str, first_name: str, reset_token: str):
        """Send password reset email"""
        try:
            subject = "Reset Your Password"
            
            # Create reset URL
            reset_url = f"http://localhost:3000/reset-password?token={reset_token}"
            
            # HTML email content
            html_content = f"""
            <html>
                <body>
                    <h2>Password Reset Request</h2>
                    <p>Hi {first_name},</p>
                    <p>You requested to reset your password. Click the link below to set a new password:</p>
                    <p><a href="{reset_url}" style="background-color: #f44336; color: white; padding: 10px 20px; text-decoration: none; border-radius: 5px;">Reset Password</a></p>
                    <p>If the button doesn't work, copy and paste this link into your browser:</p>
                    <p>{reset_url}</p>
                    <p>This link will expire in 1 hour.</p>
                    <p>If you didn't request this, please ignore this email.</p>
                    <p>Best regards,<br>E-commerce Platform Team</p>
                </body>
            </html>
            """
            
            # For development, just log the email instead of sending
            logger.info(f"Password reset email sent to {to_email}")
            logger.info(f"Reset URL: {reset_url}")
            
            return True
            
        except Exception as e:
            logger.error(f"Failed to send password reset email: {e}")
            return False

File: services/notification-service/src/services/emailService.js
const nodemailer = require('nodemailer');
const logger = require('../utils/logger');

class EmailService {
  constructor() {
    this.transporter = nodemailer.createTransporter({
      host: process.env.SMTP_HOST || 'smtp.gmail.com',
      port: process.env.SMTP_PORT || 587,
      secure: false,
      auth: {
        user: process.env.SMTP_USER || 'demo@example.com',
        pass: process.env.SMTP_PASS || 'demo_password'
      }
    });
  }

  async sendEmail(to, subject, template, data = {}) {
    try {
      const htmlContent = this.generateEmailTemplate(template, data);
      
      const mailOptions = {
        from: process.env.FROM_EMAIL || 'noreply@ecommerce.com',
        to,
        subject,
        html: htmlContent
      };

      // For demo purposes, we'll log instead of actually sending
      logger.info(`📧 Email would be sent to: ${to}`);
      logger.info(`📧 Subject: ${subject}`);
      logger.info(`📧 Template: ${template}`);
      
      // Simulate email sending
      return {
        messageId: `demo_${Date.now()}`,
        accepted: [to],
        rejected: []
      };

      // Uncomment below for actual email sending
      // const result = await this.transporter.sendMail(mailOptions);
      // logger.info(`Email sent successfully to ${to}:`, result.messageId);
      // return result;
    } catch (error) {
      logger.error('Email sending error:', error);
      throw error;
    }
  }

  generateEmailTemplate(template, data) {
    const templates = {
      welcome: `
        <div style="font-family: Arial, sans-serif; max-width: 600px; margin: 0 auto;">
          <h1 style="color: #667eea;">Welcome to Our E-commerce Store!</h1>
          <p>Hi ${data.name || 'Customer'},</p>
          <p>Thank you for joining our community! We're excited to have you on board.</p>
          <p>Start exploring our amazing products and enjoy exclusive deals.</p>
          <a href="${data.loginUrl || '#'}" style="background: #667eea; color: white; padding: 12px 24px; text-decoration: none; border-radius: 5px; display: inline-block; margin: 20px 0;">
            Start Shopping
          </a>
          <p>Best regards,<br>The E-commerce Team</p>
        </div>
      `,
      order_confirmation: `
        <div style="font-family: Arial, sans-serif; max-width: 600px; margin: 0 auto;">
          <h1 style="color: #28a745;">Order Confirmed!</h1>
          <p>Hi ${data.name || 'Customer'},</p>
          <p>Your order #${data.orderId || 'N/A'} has been confirmed and is being processed.</p>
          <div style="background: #f8f9fa; padding: 20px; border-radius: 5px; margin: 20px 0;">
            <h3>Order Details:</h3>
            <p><strong>Order ID:</strong> ${data.orderId || 'N/A'}</p>
            <p><strong>Total:</strong> $${data.total || '0.00'}</p>
            <p><strong>Estimated Delivery:</strong> ${data.deliveryDate || '3-5 business days'}</p>
          </div>
          <a href="${data.trackingUrl || '#'}" style="background: #28a745; color: white; padding: 12px 24px; text-decoration: none; border-radius: 5px; display: inline-block; margin: 20px 0;">
            Track Your Order
          </a>
          <p>Thank you for your purchase!</p>
        </div>
      `,
      password_reset: `
        <div style="font-family: Arial, sans-serif; max-width: 600px; margin: 0 auto;">
          <h1 style="color: #dc3545;">Password Reset Request</h1>
          <p>Hi ${data.name || 'Customer'},</p>
          <p>You requested to reset your password. Click the button below to create a new password:</p>
          <a href="${data.resetUrl || '#'}" style="background: #dc3545; color: white; padding: 12px 24px; text-decoration: none; border-radius: 5px; display: inline-block; margin: 20px 0;">
            Reset Password
          </a>
          <p>This link will expire in 1 hour. If you didn't request this, please ignore this email.</p>
          <p>Best regards,<br>The E-commerce Team</p>
        </div>
      `,
      shipping_update: `
        <div style="font-family: Arial, sans-serif; max-width: 600px; margin: 0 auto;">
          <h1 style="color: #17a2b8;">Shipping Update</h1>
          <p>Hi ${data.name || 'Customer'},</p>
          <p>Your order #${data.orderId || 'N/A'} has been shipped!</p>
          <div style="background: #f8f9fa; padding: 20px; border-radius: 5px; margin: 20px 0;">
            <h3>Shipping Details:</h3>
            <p><strong>Tracking Number:</strong> ${data.trackingNumber || 'N/A'}</p>
            <p><strong>Carrier:</strong> ${data.carrier || 'Standard Shipping'}</p>
            <p><strong>Estimated Delivery:</strong> ${data.estimatedDelivery || '2-3 business days'}</p>
          </div>
          <a href="${data.trackingUrl || '#'}" style="background: #17a2b8; color: white; padding: 12px 24px; text-decoration: none; border-radius: 5px; display: inline-block; margin: 20px 0;">
            Track Package
          </a>
          <p>Thank you for your patience!</p>
        </div>
      `
    };

    return templates[template] || `
      <div style="font-family: Arial, sans-serif; max-width: 600px; margin: 0 auto;">
        <h1>Notification</h1>
        <p>${data.message || 'You have a new notification.'}</p>
      </div>
    `;
  }
}

module.exports = new EmailService();
